src/core/server/integration_tests/saved_objects/service/lib/repository_with_proxy.test.ts

```diff
@@ -139,9 +139,11 @@ describe('404s from proxies', () => {
   });
 
   afterAll(async () => {
-    await root.shutdown();
-    await hapiServer.stop({ timeout: 1000 });
-    await esServer.stop();
+    if (root) {
```

Comment:
If we get to a point where we started `esServer`, but we did not initialize `root` (or `hapiServer`), we will not be explicitly stopping them here. Perhaps we could remove the `if` statement and do something like:
```
  await root?.shutdown();
  await hapiServer?.stop({ timeout: 1000 });
  await esServer?.stop();
```
WDYT?

Reply:
I see that used in other tests but prefer not to use shorthand here. It's very easy to delete a `?`, but less easy to remove a whole `if` condition.